tests/wen_transfer_guard.test.ts

```diff
@@ -0,0 +1,171 @@
+import * as anchor from "@coral-xyz/anchor";
+import { AnchorProvider, Program, web3 } from "@coral-xyz/anchor";
+
+import { WenTransferGuard } from "../target/types/wen_transfer_guard";
+import {
+  ExtensionType,
+  ExtraAccountMeta,
+  TOKEN_2022_PROGRAM_ID,
+  createAssociatedTokenAccountInstruction,
+  createInitializeMintInstruction,
+  createInitializeTransferHookInstruction,
+  createMintToInstruction,
+  getAssociatedTokenAddressSync,
+  getExtraAccountMetaAddress,
+  getMintLen,
+  addExtraAccountMetasForExecute,
+  createTransferCheckedInstruction,
+} from "@solana/spl-token";
+
+describe("wen_transfer_guard", () => {
+  const provider = anchor.AnchorProvider.env();
+  anchor.setProvider(provider);
+  const { connection, wallet } = provider;
+  const program = anchor.workspace.WenNewStandard as Program<WenTransferGuard>;
+
+  const mintAuthority = web3.Keypair.generate();
+  const payer = web3.Keypair.generate();
+  const mint = web3.Keypair.generate();
+  const sourceAuthority = web3.Keypair.generate();
+  const destinationAuthority = web3.Keypair.generate().publicKey;
+
+  let source: web3.PublicKey = null;
+  let destination: web3.PublicKey = null;
+
+  let extraMetasAddress: web3.PublicKey = null;
+
+  const decimals = 2;
+  const mintAmount = 100;
+  const transferAmount = 10;
+
+  const extraMetas: ExtraAccountMeta[] = [
+    {
+      discriminator: 0,
+      addressConfig: web3.Keypair.generate().publicKey.toBuffer(),
```

Comment:
```suggestion
      guard: web3.Keypair.generate().publicKey.toBuffer(),
```

Reply:
The one account we want to register with extra account meta is the guard.

Reply:
Do we want to abstract this into `guard`?
This translates directly into a `spl_tlv_account_resolution::account::ExtraAccountMeta` account type, using the same fields.

Reply:
Initially no because many mints will assign the same guard so it cant store the bump for the extra account meta.